crates/test-utils/src/execution/contract.rs

```diff
@@ -0,0 +1,116 @@
+use std::fs;
+use std::path::Path;
+
+use ethers::abi::Tokenize;
+use ethers_solc::artifacts::CompactContractBytecode;
+use foundry_config::{find_project_root_path, load_config};
+use kakarot_rpc_core::client::constants::CHAIN_ID;
+use kakarot_rpc_core::models::felt::Felt252Wrapper;
+use reth_primitives::{Transaction, TransactionKind, TxEip1559};
+use starknet_crypto::FieldElement;
+
+use crate::root_project_path;
+
+pub trait EvmContract {
+    fn load_contract_bytecode(contract_name: &str) -> Result<CompactContractBytecode, eyre::Error> {
+        let dot_sol = format!("{contract_name}.sol");
+        let dot_json = format!("{contract_name}.json");
+
+        let foundry_default_out = load_config().out;
+        let compiled_solidity_relative_path = Path::new(&foundry_default_out).join(dot_sol).join(dot_json);
+        let compiled_solidity_global_path = root_project_path!(&compiled_solidity_relative_path);
+
+        let compiled_solidity_file_content = fs::read_to_string(compiled_solidity_global_path)?;
+        Ok(serde_json::from_str(&compiled_solidity_file_content)?)
+    }
+
+    fn create_transaction<T: Tokenize>(
+        contract_bytecode: &CompactContractBytecode,
+        constructor_args: T,
+        nonce: u64,
+    ) -> Result<Transaction, eyre::Error>;
+
+    fn call_transaction<T: Tokenize>(
+        &self,
+        selector: &str,
+        constructor_args: T,
+        nonce: u64,
+        value: u128,
+    ) -> Result<Transaction, eyre::Error>;
+}
+
+#[derive(Default)]
+pub struct KakarotEvmContract {
+    pub bytecode: CompactContractBytecode,
+    pub starknet_address: FieldElement,
+    pub evm_address: FieldElement,
+}
+
+impl KakarotEvmContract {
+    pub fn new(bytecode: CompactContractBytecode, starknet_address: FieldElement, evm_address: FieldElement) -> Self {
+        Self { bytecode, starknet_address, evm_address }
+    }
+}
+
+impl EvmContract for KakarotEvmContract {
+    fn create_transaction<T: Tokenize>(
```

Comment:
this is a bit ambiguous with the idea of "creating" a transaction..
thinking out loud here, let's add a prefix "prepare"?

`prepare_create_tx` and 
`prepare_call_tx`?

Reply:
yeah I agree, will update